Sources/Sentry/SentryTransportFactory.m

```diff
@@ -27,6 +29,13 @@ @implementation SentryTransportFactory
                                sentryFileManager:(SentryFileManager *)sentryFileManager
                                       rateLimits:(id<SentryRateLimits>)rateLimits
 {
+    if (!options.parsedDsn) {
+        SENTRY_LOG_FATAL(@"Failed to create transports because the SentryOptions does not contain "
+                         @"a parsed DSN.");
+        return @[];
+    }
+    SentryDsn *_Nonnull dsn = SENTRY_UNWRAP_NULLABLE(SentryDsn, options.parsedDsn);
```

Comment:
`m`: The Spotlight transport doesn't require a DSN. You could use it without a DSN. Currently, the client drops everything if no DSN is set. So ideally, we would allow this in the client, but we don't need to change this now. Instead, we should only log a warning or even only a debug. I also think some users set a wrong DSN on purpose to avoid sending any data. The fatal error message could be confusing to some users I believe. Also we already log an error message somewhere else if we couldn't parse the DSN.